students/Yue_Ma/Lesson03/mailroom.py

```diff
@@ -54,32 +50,33 @@ def send_a_thank_you_note():
                 donor_list[user_name] = list_history
                 print('Updated the info to our database.')
                 print(email(user_name))
-                back_to_main()
             else:
                 print('')
                 print('This name is NOT in our database!')
                 donor_list[user_name] = [float(user_amount)]
                 print('Added the name to our database.')
                 print(email(user_name))
-                back_to_main()
-            break
         elif user_send_a_thank_you_note == '2':
-            back_to_main()
             break
         else:
             print("Not a valid option!!!")
+    return main()
 
 
 def create_a_report():
     title = {'Donor Name': ['Total Given', 'Num Gifts', 'Average Gift']}
     value_title = list(title.values())
     key_title = list(title.keys())
 
+    # sort
+    donor_list_sort = dict(sorted(donor_list.items(), key=lambda t: t[0]))
+    print(donor_list_sort)
+
     # calculate total donation
     donation_matrix = []
-    for j in range(0, len(donor_list)):
-        donation_matrix.append([list(donor_list.keys())[j], sum(list(donor_list.values())[j]),
-                                len(list(donor_list.values())[j])])
+    for j in range(0, len(donor_list_sort)):
+        donation_matrix.append([list(donor_list_sort.keys())[j], sum(list(donor_list_sort.values())[j]),
+                                len(list(donor_list_sort.values())[j])])
```

Comment:
this function looks a bit complicated, not sure why you're generating a dictionary?

a somewhat simpler version may look like this:
```
sorted_list = sorted(donor_list, key=total_donated, reverse=True)
for item in sorted_list:
    print("...")
```
```